src/Symfony/Component/HttpClient/PluggableHttpClient.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Component\HttpClient\Response\PluggableResponse;
+use Symfony\Component\HttpClient\Response\ResponseStream;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\HttpClient\ResponseStreamInterface;
+
+/**
+ * Allows processing responses while streaming them.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class PluggableHttpClient implements HttpClientInterface
+{
+    private $client;
+    private $pluggableResponseFactory;
+
+    public function __construct(HttpClientInterface $client, callable $pluggableResponseFactory)
```

Comment:
It's possible to give an array as callable (`[$this, 'method']`) with this signature?

Reply:
Yes: https://3v4l.org/Xq5hY